hack/tools.go

```diff
@@ -15,6 +15,7 @@ import (
 	_ "github.com/gardener/gardener/hack/tools/report-collector"
 
 	_ "github.com/ahmetb/gen-crd-api-reference-docs"
+	_ "go.uber.org/mock/gomock"
```

Comment:
What is the reason this package to be imported here?

Reply:
All tools that are required need to be pulled forcefully ref- 
https://github.com/gardener/gardener/blob/2437fcf30f80ffecedb4b823551d81c0c91fc69c/hack/tools.go#L18-L33

Reply:
But `github.com/golang/mock` was never a dependency of the `hack` module, why does the replacement `go.uber.org/mock/gomock` is now made a dependency?

Reply:
Yes, it was not, `go.uber.org/mock/gomock` can also be removed as `go.uber.org/mock/gomock` is pulled anyways by other packages.

Reply:
Removed it in 9e408b0